### PR TITLE
fixmes be gone part 2

### PR DESCRIPTION
**Pull request by MabezDev**
## Thank you for your contribution!

We appreciate the time and effort you've put into this pull request.
To help us review it efficiently, please ensure you've gone through the following checklist:

### Submission Checklist 📝
- [x] I have updated existing examples or added new ones (if applicable).
- [x] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
#### Extra:
- [x] I have read the [CONTRIBUTING.md guide](https://github.com/esp-rs/esp-hal/blob/main/CONTRIBUTING.md) and followed its instructions.

### CODE REVIEW

File: esp-hal/src/analog/adc/esp32.rs [hunk 1]
```diff
@@ -194,9 +194,7 @@ impl RegisterAccess for ADC2 {
 /// Analog-to-Digital Converter peripheral driver.
 pub struct ADC<'d, ADC> {
     _adc: PeripheralRef<'d, ADC>,
-    #[allow(dead_code)] // FIXME
```
Comment (MabezDev) [inline esp-hal/src/analog/adc/esp32.rs:197]: This was already fixed by removing the dep on the eh2 deps, thanks @jessebraham :sunglasses:

File: esp-hal/src/rtc_cntl/rtc/esp32c6.rs [hunk 2]
```diff
@@ -303,7 +303,6 @@ fn modem_lpcon_ll_set_wifi_lpclk_divisor_value(divider: u16) {
 }
 
 fn modem_clock_hal_enable_wifipwr_clock(enable: bool) {
-    // FIXME: esp-idf uses refcounting here for later revisions.
```
Comment (MabezDev) [inline esp-hal/src/rtc_cntl/rtc/esp32c6.rs:306]: Created https://github.com/esp-rs/esp-hal/issues/1447 to track this globally

**Review by SergioGasquez — APPROVED** [review]
LGTM, thanks!

File: esp-hal/src/soc/esp32h2/gpio.rs [hunk 3]
```diff
@@ -256,7 +256,6 @@ pub enum OutputSignal {
     GPIO             = 128,
 }
 
-// FIXME: add alternate function numbers/signals where necessary
```
Comment (MabezDev) [inline esp-hal/src/soc/esp32h2/gpio.rs:259]: To the best of my current knowledge, this looked complete.

**Review by jessebraham — APPROVED** [review]
Thanks!

**Pull request merged**